### PR TITLE
Parent mmr root & implement api error response

### PR DESCRIPTION
**Pull request by clearloop**
companion of [bridger#95](https://github.com/darwinia-network/bridger/pull/95)

1. Replace the default response with error messages
2. Api mmr_root changed to parent_mmr_root

### CODE REVIEW

**Comment by HackFisher** [comment]
@wuminzhe Will need a companion PR for bridger to handle the new web result (error handling), would you please help provide one, thanks.

**Review by wuminzhe — APPROVED** [review]

File: src/api/ethereum/helper.rs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+};
+use actix_web::error;
+use cmmr::MMR;
+use primitives::{chain::ethereum::EthereumHeaderJson, rpc::RPC};
+
+/// Web result
+pub type WebResult<R> = Result<R, error::Error>;
+
+/// Get mmr_root string with web response
+pub fn mmr_root(block: u64, shared: &ShadowShared) -> WebResult<String> {
```
Comment (HackFisher) [inline src/api/ethereum/helper.rs:14]:
```suggestion
pub fn parent_mmr_root(block: u64, shared: &ShadowShared) -> WebResult<String> {
```

File: src/api/ethereum/helper.rs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    ShadowShared,
+};
+use actix_web::error;
+use cmmr::MMR;
+use primitives::{chain::ethereum::EthereumHeaderJson, rpc::RPC};
+
+/// Web result
+pub type WebResult<R> = Result<R, error::Error>;
+
+/// Get mmr_root string with web response
```
Comment (HackFisher) [inline src/api/ethereum/helper.rs:13]:
```suggestion
/// Get parent_mmr_root string with web response
```

**Review by HackFisher — CHANGES_REQUESTED** [review]

File: src/api/ethereum/mmr_leaf.rs [hunk 3]
```diff
@@ -47,15 +45,24 @@ impl Into<MMRLeafJson> for MMRLeaf {
...
     let num: u64 = block.to_string().parse().unwrap_or(0);
-    let leaf = H256::hex(
-        &(&shared.store)
-            .get_elem(cmmr::leaf_index_to_pos(num))
-            .unwrap().unwrap_or([0;32]),
-    );
+    let leaf = (&shared.store)
+        .get_elem(cmmr::leaf_index_to_pos(num))
+        .unwrap_or_default()
+        .unwrap_or([0; 32]);
```
Comment (HackFisher) [inline src/api/ethereum/mmr_leaf.rs:56]: Using match to return approximate result

**Review by HackFisher — CHANGES_REQUESTED** [review]

**Review by HackFisher — APPROVED** [review]

**Pull request merged**